extern/sector-storage/fsutil/filesize_unix.go

```diff
@@ -11,19 +12,29 @@ type SizeInfo struct {
 	OnDisk int64
 }
 
-// FileSize returns bytes used by a file on disk
+// FileSize returns bytes used by a file or directory on disk
+// NOTE: We care about the allocated bytes, not file or directory size
 func FileSize(path string) (SizeInfo, error) {
-	var stat syscall.Stat_t
-	if err := syscall.Stat(path, &stat); err != nil {
-		if err == syscall.ENOENT {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			stat := info.Sys().(*syscall.Stat_t)
```

Comment:
I'd still defensively check this. Failing to check storage space is much less bad than crashing the miner process and potentially missing PoSts
```suggestion
			stat, ok := info.Sys().(*syscall.Stat_t)
			if !ok {
				return xerrors.New("FileInfo.Sys of wrong type")
			}
```

Reply:
Fair enough!